core/src/test/scala/org/apache/spark/internal/config/ConfigEntrySuite.scala

```diff
@@ -128,6 +128,25 @@ class ConfigEntrySuite extends SparkFunSuite {
     assert(conf.get(transformationConf) === "bar")
   }
 
+  test("conf entry: checkValue()") {
+    def createConf(default: Int): ConfigEntry[Int] =
+      ConfigBuilder(testKey("checkValue"))
+        .intConf
+        .checkValue(value => value >= 0, "value must be non-negative")
+        .createWithDefault(default)
+
+    // this succeeds
+    val conf = createConf(10)
+
+    // this fails because valueConverter() calls checkValue()
+    val e1 = intercept[IllegalArgumentException] { conf.valueConverter("-1") }
```

Comment:
to be safer, we should not change the existing behavior. So `conf.set` should not fail here and we can follow the next test